.tekton/operator-bundle-pipeline.yaml

```diff
@@ -658,3 +661,92 @@ spec:
       - name: kind
         value: task
       resolver: bundles
+
+  - name: create-acs-style-snapshot
+    # Explicitly running after all other tasks to ensure that
+    # - there are no failures that should prevent a release of the operator-bundle image (missing RPMs signatures, deprecated base images, ...)
+    # - the source image is present as it is required by EC
+    runAfter:
+    - apply-tags
+    - build-source-image
+    - clair-scan
+    - clamav-scan
+    - deprecated-base-image-check
+    - rpms-signature-scan
+    - sast-snyk-check
+    - push-dockerfile
```

Comment:
> It'll be a pain to maintain this list... Do you have any ideas how can we keep it up to date as the time passes and tasks get added or changed?

No, I will think about it. A non-technical solution (ie a comment at the top of the file: "if you add tasks, ... update the runAfter" might suffice.

> One option to try would be to just move create-acs-style-snapshot to finally:. What I wonder is how the failure of it would look like and whether we can still receive Slack messages for that.

all `finally` tasks run in parallel, ie we would not get a Slack notification if the snapshot creation failed. For example, if the pipeline times out waiting for an image, it cannot create a snapshot and would fail this task. I don't think we should move it to `finally`.

Reply:
> all finally tasks run in parallel

Bummer

> A non-technical solution (ie a comment at the top of the file: "if you add tasks, ... update the runAfter" might suffice.

This will fall apart rather quickly because most of times engineer scroll directly to the place they need to change without bothering to re-read comments at the top each time.

One option I see is to create a validation script that would use `yq` to query task names and check that all are listed in `runAfter` and put this script to run in CI.

Alternatively, I'd suggest to sacrifice Slack notification for failing Snapshot creation and move the Snapshot creation to `finally`. Still, a failure there would fail the pipeline so we should notice.

Can't see other options.

Reply:
> One option I see is to create a validation script that would use yq to query task names and check that **all** are listed in runAfter and put this script to run in CI.

By "all", do you mean each and every task under `.spec.tasks` except `create-acs-style-snapshot`? If you mean just the tasks currently listed in the `runAfter`, we would re-implement a pipeline parser as we'd need to build the pipeline graph and understand which tasks are on which level.

Reply:
A suggestion done in https://github.com/stackrox/stackrox/pull/13577/commits/a245624c223f482d05e3268e872fc647d2dcc436.

Reply:
Proof: https://github.com/stackrox/stackrox/actions/runs/12415215130/job/34661075839?pr=13577

Reply:
<img width="916" alt="Screenshot 2024-12-20 at 11 35 29" src="https://github.com/user-attachments/assets/b5ede07b-430e-4a81-8016-60ffba383668" />

Done what we discussed in https://github.com/stackrox/stackrox/pull/13577/commits/a10584c9626485c342a550a482ebe19ddc0b848c